apollo-router/src/services/query_planner.rs

```diff
@@ -36,9 +38,41 @@ impl Request {
     pub(crate) fn new(
         query: String,
         operation_name: Option<String>,
-        compiler: Arc<Mutex<ApolloCompiler>>,
         context: Context,
+        compiler: Arc<Mutex<ApolloCompiler>>,
     ) -> Request {
+        Self {
+            query,
+            operation_name,
+            context,
+            compiler,
+        }
+    }
+}
+
+/// [`Context`] for the request.
+#[derive(Clone, Derivative)]
+#[derivative(Debug)]
+pub(crate) struct CachingRequest {
```

Comment:
before merging #3200, the caching request did not have the compiler. But it's better to keep the types separated, because at some point, `CachingRequest` will have an additional field to change the cache key